src/ceph-disk/ceph_disk/main.py

```diff
@@ -2545,6 +2567,60 @@ def get_dmcrypt_type(args):
             return None
 
 
+class Secrets(object):
+
+    def __init__(self):
+        secret, stderr, ret = command(['ceph-authtool', '--gen-print-key'])
+        LOG.debug("stderr " + stderr)
+        assert ret == 0
```

Comment:
there is no need to have an assert here, it would just crash and display a traceback to the user. Instead, there is an existing function called `command_check_call` that will handle the error nicely and report it back to the user